### DISCUSSION TITLE
Construct SELECT query to select specified columns

### DISCUSSION THREAD

**Discussion by NagarajG05** [sqlalchemy/sqlalchemy#11239] [Usage Questions]
I am trying to construct select query to select on list columns provided as well want to remove table prefix on each column when a query is generated.

here is my code

```
from sqlalchemy import create_engine, MetaData, Table, cast, DATE, TIMESTAMP, select,text,column
engine = create_engine(f'hana://{user}@{host}/{db}')


metadata = MetaData()
table = Table(table_name, metadata, autoload=True, autoload_with=engine,schema=schema_name)

print(select(table))
```

above code is displaying query as:

```
select 
schema.table.col1,  schema.table."Col1", schema.table."col3"
 from schema.table

```

but I am looking for something like :
```
select 
col1,  "Col1", "col3"
 from schema.table
```

additionally i need to generate query for specified columns. e g.: in case if want to select only col1 and col3 then query should be
`select col1, "col3"  from schema.table`

any inputs on this will be very helpful

Thank you

**Comment by CaselIT** [comment]
Hi,

Make sure that the table name doesn't contain the schema, that should be specific using the schema kw.
It it's already like that, then This seems like a bug with the hana dialect, since it's using the schema as part of the table name.

You should report to their maintainers

**Reply by NagarajG05** [reply]
Thanks for the reply.
 
Generated select query is having table name appended to column along with schema. I just want to have column name without table/schema appended

**Reply by CaselIT** [reply]
It think sqlalchemy by default will add table plus column name, but that can be changed

The issue here is that the table name seems to include the schema

**Comment by NagarajG05** [comment]
I am able to get rid of table and schema from appending to column. Here is the code
```

identifier_preparer = engine.dialect.identifier_preparer
columns = []
for column in table.columns:
    column_name = identifier_preparer.format_column(column)
    columns.append(literal_column(column_name))
select_query = select(columns).select_from(table)
```

But I am not able to generate query to select specific columns
